akka-actor-typed/src/main/scala/akka/actor/typed/internal/entity/EntityProvider.scala

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (C) 2021 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.actor.typed.internal.entity
+
+import akka.actor.typed.ActorRef
+import akka.actor.typed.Entity
+import akka.actor.typed.EntityRef
+import akka.actor.typed.EntityTypeKey
+import akka.annotation.InternalApi
+
+/**
+ * Marker interface to use with dynamic access
+ *
+ * INTERNAL API
+ */
+@InternalApi
+private[akka] trait EntityProvider {
+
+  def initEntity[M, E](entity: Entity[M, E]): ActorRef[E]
+
+  def entityRefFor[M](typeKey: EntityTypeKey[M], entityId: String): EntityRef[M]
+}
+
+private[akka] object LocalEntityProvider extends EntityProvider {
```

Comment:
I think it would be worth having a new implementation for the local case. It would be much simpler than the cluster variant. No shards at all. Still handling lifecycle and passivation.

The passivation strategies that Peter is working on can probably be used for the local case also.